src/sql/workbench/api/node/mainThreadConnectionManagement.ts

```diff
@@ -43,6 +43,19 @@ export class MainThreadConnectionManagement implements MainThreadConnectionManag
 		this._toDispose = dispose(this._toDispose);
 	}
 
+	public $getConnections(activeConnectionsOnly?: boolean): Thenable<azdata.connection.ConnectionProfile[]> {
+		let connections = this._connectionManagementService.getActiveConnections();
```

Comment:
I'd like to see logic this in the connectionManagementService actually  - that way we can internally use the service directly